src/printer.rs

```diff
@@ -315,17 +314,60 @@ impl<'a> Printer for InteractivePrinter<'a> {
         };
 
         let description = &input.description;
+        let metadata = &input.metadata;
+
+        // We use this iterator to have a deterministic order for
+        // header components. HashSet has arbitrary order, but Vec is ordered.
+        let header_components: Vec<StyleComponent> = [
+            (
+                StyleComponent::HeaderFilename,
+                self.config.style_components.header_filename(),
+            ),
+            (
+                StyleComponent::HeaderFilesize,
+                self.config.style_components.header_filesize(),
+            ),
+        ]
+        .iter()
+        .filter(|(_, is_enabled)| *is_enabled)
+        .map(|(component, _)| *component)
+        .collect();
+
+        // Print the cornering grid before the first header component
+        if self.config.style_components.grid() {
+            self.print_horizontal_line(handle, '┬')?;
+        } else {
+            // Only pad space between files, if we haven't already drawn a horizontal rule
+            if add_header_padding && !self.config.style_components.rule() {
+                writeln!(handle)?;
+            }
+        }
 
-        writeln!(
-            handle,
-            "{}{}{}",
-            description
-                .kind()
-                .map(|kind| format!("{}: ", kind))
-                .unwrap_or_else(|| "".into()),
-            self.colors.filename.paint(description.title()),
-            mode
-        )?;
+        header_components.iter().try_for_each(|component| {
+            self.print_header_component_indent(handle)?;
+
+            match component {
+                StyleComponent::HeaderFilename => writeln!(
+                    handle,
+                    "{}{}{}",
+                    description
+                        .kind()
+                        .map(|kind| format!("{}: ", kind))
+                        .unwrap_or_else(|| "".into()),
+                    self.colors.header_value.paint(description.title()),
+                    mode
+                ),
+
+                StyleComponent::HeaderFilesize => {
+                    let bsize = metadata
+                        .size
+                        .map(|s| format!("{}", ByteSize(s)))
+                        .unwrap_or("".into());
```

Comment:
It's quite possible that `metadata.size` is empty, right? For example, if the input comes from STDIN. I think we should also add a test case for this. Maybe we want to skip the `Size: ` line completely in this case? Or at least print something like `Size: -`.